File: Java/O(1)/o1_2.java
// Java program for the above approach
//https://www.geeksforgeeks.org/java-program-to-find-the-mth-element-of-the-array-after-k-left-rotations/
import java.util.*;
class on2{
	
// Function to return Mth element of
// array after k left rotations
public static int getFirstElement(int[] a, int N,
								int K, int M)
{
	
	// The array comes to original state
	// after N rotations
	K %= N;

	// Mth element after k left rotations
	// is (K+M-1)%N th element of the
	// original array
	int index = (K + M - 1) % N;

	int result = a[index];

	// Return the result
	return result;
}

// Driver code
public static void main(String[] args)
{
	
	// Array initialization
	int a[] = { 3, 4, 5, 23 };

	// Size of the array
	int N = a.length;

	// Given K rotation and Mth element
	// to be found after K rotation
	int K = 2, M = 1;

	// Function call
	System.out.println(getFirstElement(a, N, K, M));
}
}

// This code is contributed by divyeshrabadiya07

File: Python/O(1)/o1_2.py
# Python3 program to implement
# the above approach

#https://www.geeksforgeeks.org/python3-program-to-find-mth-element-after-k-right-rotations-of-an-array/
# Function to return Mth element of
# array after k right rotations
def getFirstElement(a, N, K, M):

	# The array comes to original state
	# after N rotations
	K %= N

	# If K is greater or equal to M
	if (K >= M):

		# Mth element after k right
		# rotations is (N-K)+(M-1) th
		# element of the array
		index = (N - K) + (M - 1)

	# Otherwise
	else:

		# (M - K - 1) th element
		# of the array
		index = (M - K - 1)

	result = a[index]

	# Return the result
	return result

# Driver Code
if __name__ == "__main__":
	
	a = [ 1, 2, 3, 4, 5 ]
	N = len(a)

	K , M = 3, 2

	print( getFirstElement(a, N, K, M))

# This code is contributed by chitranayal
